### DISCUSSION TITLE
authentication with js

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pocketbase/pocketbase#3047] [Q&A] [answered]
i am creating a function that will login the user if user already exists or create new one. i am using below JS code

```js
onRecordBeforeAuthWithPasswordRequest((e) => {
  if (e.record == null) {
    const collection = $app.dao().findCollectionByNameOrId("USERS");
    const record = Record(collection);
    const form = RecordUpsertForm($app, record);
    form.loadData({
      username: e.identity,
      name: e.identity,
      password: e.password,
      passwordConfirm: e.password,
    });
    form.submit();
  }
  return null;
}, "USERS");
```

but the problem is it's not waiting until a new user gets created.
for first time it's showing
`{
  "code": 400,
  "message": "Failed to authenticate.",
  "data": {}
}`
if I request again then it will give me the credentials  response 
what I want is it have to wait until the user get created and give credentials back

**Comment by ganigeorgiev** [comment, marked as answer]
You need to reasign the e.record to the newly created one, aka. `e.record = record` (or just pass `e.record = new Record(collection);` to the form).
